tiledb/sm/filesystem/gcs.cc

```diff
@@ -752,11 +807,126 @@ Status GCS::flush_object(const URI& uri) {
     return flush_object_direct(uri);
   }
 
-  LOG_FATAL("Unreached -- TODO");
+  Buffer* const write_cache_buffer = get_write_cache_buffer(uri.to_string());
+
+  const Status flush_write_cache_st =
+      flush_write_cache(uri, write_cache_buffer, true);
+
+  // We do not need to protect 'multi_part_upload_states_' here because
+  // 'count' is thread-safe.
+  if (multi_part_upload_states_.count(uri.to_string()) == 0) {
```

Comment:
Is `uri` thread-unique, here? Otherwise could you elaborate why this is thread-safe?

Reply:
Good catch, thank you. This is not thread-safe if a user is flushing obj1 while writing obj2. I'll update this here and in Azure. This is a vestige from the same assumption I was making for yesterday's bug. I'll update this here and check Azure for a similar bug.